merkletree/pad.go

```diff
@@ -95,10 +95,24 @@ func (pad *PAD) Set(name string, value []byte) error {
 	return pad.tree.Set(index, name, value)
 }
 
-func (pad *PAD) Lookup(name string) (*AuthenticationPath, error) {
+// LookupInCurrentEpoch searchs the requested name in the current tree
+// (which is not committed to the PAD).
+// The returned AuthenticationPath should be used only to verify the inclusion
+// the of requested name in the server-side
+// and should **not** be used to return to the client.
+func (pad *PAD) LookupInCurrentEpoch(name string) (*AuthenticationPath, error) {
```

Comment:
> and I'm afraid that a developer will misuse this function (accidentally or intentionally) despite your description in the documentation.

I feel that too.

> Is there an alternative way we would keep track of the names that have been registered in the current epoch but aren't committed yet (e.g. in an array)?

Let's see what @arlolra  and @Liamsi  say. I'm not so sure about this :(

> Or could merkletree.Set() return an error if we attempt to add a private index that already exists?

Actually I think I want to keep it for key change operation, and we should just do this in PAD. What do you think?